src/effects/RealtimeEffectState.cpp

```diff
@@ -144,8 +138,6 @@ bool RealtimeEffectState::AddTrack(int group, unsigned chans, float rate)
 
       // Add a NEW processor
       mEffect->RealtimeAddProcessor(gchans, rate);
-
-      // Bump to next processor
       mCurrentProcessor++;
```

Comment:
I am not sure that this scheme of things will work when having multiple effects (multiple RealtimeEffectState-s) referencing the same EffectProcessor.
Let's imagine a mono track having two instances of BassAndTreble on it. `mCurrentProcessor` is basically an index referencing the "slaves" created in BassAndTreble::RealtimeAddProcessor. I understand it like this: during this method, `mCurrentProcessor` will take a value of 0 and possibly also 1, if the track is stereo - because in that case, two slaves are needed, one per channel. The only instance of BassAndTreble will create two slaves, and this instance of RealtimeEffectState will reference them with indexes 0 and 1 - all good so far.

But what will happen when the second instance of RealtimeEffectState will reach this method? it will make the only instance of BassAndTreble create two more slaves, but it will reference them again with indexes 0 and 1, instead of 2 and 3.

This potential erratic behavior can not be proven now because AFAIK the GUI still does not allow us to place more effects on the same track; moreover, this way of managing per-instance states is going to be rewritten anyway, so I do not know if it is useful to correct this problem. 

If we need to, my proposed solution is simply that `::RealtimeAddProcessor` would return the index of the created slave, and we use that instead of the increased `mCurrentProcessor`

Reply:
Anyway - adding effects per-track is not supported by the GUI so this bug has no chance to manifest itself - plus, we will do a rewrite of how states are stored, probably before the supporting GUI code will be written; therefore, I will mark this conversation as resolved.